test_tableslurp.sh

```diff
@@ -71,87 +69,84 @@ BACKUP_BUCKET="${BUCKET}"
-	echo "SUCCESS: INDEX_FILE has been copied to S3."
+	echo "${SUCCESS} copied ${INDEX_FILE} to S3"
 else
-	echo "FAILURE: INDEX_FILE has NOT been copied to S3."
+  die "${FAILURE} copy ${INDEX_FILE} to S3"
 fi
 echo
 
-# Get/Retrieve/Download file(s) from S3 bucket on AWS from this directory:
+# Upload/Download file to/from S3 bucket on AWS from this directory:
 # s3://kimlew-tablesnap-tests/tester:/tmp/tablesnap-test/test_aws_s3_cp.txt
-
-# Run tableslurp with current & extra 4 args. Then do a check.
-# --name tester - is the flag & the replacement container ID, "${CONTAINER}"
-# index_file is:    index.txt
-# index_bucket is:  kimlew-tablesnap-tests
-# backup_bucket is: kimlew-tablesnap-tests
-
 # Rename the local file with mv, to same directory as prep for download from AWS.
-# Check removal. Run tableslurp. Check if files are the same locally, to prove
-# successful file download.
-echo "SRC_LOCAL_UPDOWN_FILE:\t" "${SRC_LOCAL_UPDOWN_FILE}"
+# Check removal.
+echo "SRC_LOCAL_UPDOWN_FILE:\t\t" "${SRC_LOCAL_UPDOWN_FILE}"
 echo "SRC_LOCAL_ORIG_TEST_FILE:\t" "${SRC_LOCAL_ORIG_TEST_FILE}"
 mv "${SRC_LOCAL_UPDOWN_FILE}" "${SRC_LOCAL_ORIG_TEST_FILE}"
-echo
+ls -lah "${DIR_LOCAL}"
 
-# Check if test file is temporarily removed from /tmp/tablesnap-test.
-if [ -f "${SRC_LOCAL_UPDOWN_FILE}" ]; then
-  echo "${UPDOWN_FILE} was NOT deleted. Directory has:"
-  ls -lah "${DIR_LOCAL}"
+# Check if test file is temporarily removed from /tmp/tablesnap-test"
+if [ ! -f "${SRC_LOCAL_UPDOWN_FILE}" ]; then
+  echo "${SUCCESS} deleted ${UPDOWN_FILE}"
 else
-  echo "${UPDOWN_FILE} has been deleted. Directory now has:"
-  ls -lah "${DIR_LOCAL}"
+  die "${FAILURE} delete ${UPDOWN_FILE}"
 fi
 echo
-echo
 
+# Run tableslurp with current & extra 4 args. Check if files are the same locally.
+# --name tester - is the flag & the replacement container ID, "${CONTAINER}"
+# INDEX_FILE is:    index.txt
+# INDEX_BUCKET is:  kimlew-tablesnap-tests
+# BACKUP_BUCKET is: kimlew-tablesnap-tests
 ./${PYTHON_ENV}/bin/python ./tableslurp -k "$AWS_ACC_KEY_ID" -s "$AWS_SEC_ACC_KEY" --name "${CONTAINER}" "${INDEX_BUCKET}" "${INDEX_FILE}" "${BACKUP_BUCKET}"
 echo
-echo "Running tableslurp."
-echo "UpDown file has contents: "
-cat "${SRC_LOCAL_UPDOWN_FILE}"
-echo "Orig file has contents: "
-cat "${SRC_LOCAL_ORIG_TEST_FILE}"
-
-# Check if test_aws_s3_cp.txt was successfully downloadeded locally to
-# /tmp/tablesnap-test. If ! diff - which means diff returns something, there is
-# a Error. SUCCESS when diff runs & returns exit 0/returns nothing, i.e. there
-# is NO difference between the files so contents match between uploaded &
-# downloaded files.
+echo "Running tableslurp..."
+echo
+
+# Check if test_aws_s3_cp.txt was successfully downloaded locally to
+# /tmp/tablesnap-test. Note: if ! diff - means there IS a difference between
+# the 2 files, i.e., returns a status code of 1, & there IS an ERROR. If
+# ! diff returns a status code of 0, there is NO difference between the files,
```

Comment:
Did you mean `If diff returns a status code of 0, there is NO difference...`?

Reply:
I found it takes too much time to remember what diff actually returns, which requires you to remember it returns a status code of 0 or 1 and then you have to figure out the negation part. 
As an aside, 'cmp' acts the same way as 'diff' so these 2 lines return the same results, e.g., returns status code 1 so is an ERROR:

'''if ! diff "${SRC_LOCAL_UPDOWN_FILE}" "${SRC_LOCAL_ORIG_TEST_FILE}"; then'''
'''if ! cmp --silent "${SRC_LOCAL_UPDOWN_FILE}" "${SRC_LOCAL_ORIG_TEST_FILE}"; then'''

As you suggested, I used a case statement instead. I think this makes it faster to see that diff returns a status code & what the status code means.

Reply:
Case statement looks good.